src/screens/FilterTransactionsScreen.tsx

```diff
@@ -26,24 +29,62 @@ enum Index {
 interface Props {
   navigation: StackNavigationProp<MainCardStackNavigatorParams, Route.FilterTransactions>;
   route: RouteProp<MainCardStackNavigatorParams, Route.FilterTransactions>;
+  filters: Filters;
+  updateFilters: (filters: Filters) => UpdateFiltersAction;
+}
+
+interface State {
+  derived: boolean;
+  address: string;
+  dateKey: number;
+  isCalendarVisible: boolean;
+  fromDate: string;
+  toDate: string;
+  fromAmount: string;
+  toAmount: string;
+  transactionType: string;
+  transactionStatus: string;
 }
 
 const transactionStatusList = [TxType.RECOVERY, TxType.ALERT_PENDING, TxType.ALERT_CONFIRMED, TxType.ALERT_RECOVERED];
 
-export const FilterTransactionsScreen = (props: Props) => {
-  const [address, setAddress] = useState('');
-  const [dateKey, setDateKey] = useState(0);
-  const [isCalendarVisible, setIsCalendarVisible] = useState(false);
-  const [fromDate, setFromDate] = useState('');
-  const [toDate, setToDate] = useState('');
-  const [fromAmount, setFromAmount] = useState('');
-  const [toAmount, setToAmount] = useState('');
-  const [transactionType, setTransactionType] = useState(CONST.receive);
-  const [transactionStatus, setTransactionStatus] = useState('');
-  const { onFilterPress } = props.route.params;
-
-  const onFilterButtonPress = () => {
-    onFilterPress({
+class FilterTransactionsScreen extends PureComponent<Props, State> {
+  state = {
+    derived: false,
+    address: '',
+    dateKey: 0,
+    isCalendarVisible: false,
+    fromDate: '',
+    toDate: '',
+    fromAmount: '',
+    toAmount: '',
+    transactionType: CONST.receive,
+    transactionStatus: '',
+  };
+
+  static getDerivedStateFromProps(props: Props, state: State) {
```

Comment:
We should just read filters from redux. All the filter updates should update the redux state, not the local.

Reply:
Why? We update the global state when the user chooses all filters when pressing the button. Do you want to update the global state on typing for example in amount input? It looks like overkill.